core/src/solution_submission.rs

```diff
@@ -155,19 +159,48 @@ impl<'a> StableXSolutionSubmitting for StableXSolutionSubmitter<'a> {
         solution: Solution,
         claimed_objective_value: U256,
     ) -> BoxFuture<Result<(), SolutionSubmissionError>> {
+        const GAS_PRICE_CAP: u64 = 60_000_000_000;
         async move {
-            match retry_with_gas_price_increase(
+            let submit_future = retry_with_gas_price_increase(
```

Comment:
I think `submit_solution` method should take a `nonce` parameter, and we should use `web3` here to read it. That way the `send_noop_transaction` is guaranteed to have the same nonce as the solution submission transaction.

I would be OK with adding that in a follow up PR though.